Lib/test/libregrtest/main.py

```diff
@@ -115,32 +115,36 @@ def get_executed(self):
                 | set(self.run_no_tests))
 
     def accumulate_result(self, result, rerun=False):
-        test_name = result.name
+        test_name = result.test_name
 
-        if not isinstance(result, (ChildError, Interrupted)) and not rerun:
-            self.test_times.append((result.duration_sec, test_name))
+        if not(result.state == State.CHILD_ERROR or result.state == State.INTERRUPTED) and not rerun:
+            self.test_times.append((result.duration, test_name))
 
-        if isinstance(result, Passed):
+        state = result.state
```

Comment:
Pattern-matching could improve the readability of this block, too (the diff is relative to your PR branch):

```diff
diff --git a/Lib/test/libregrtest/main.py b/Lib/test/libregrtest/main.py
index b0de7f879c..0616778adc 100644
--- a/Lib/test/libregrtest/main.py
+++ b/Lib/test/libregrtest/main.py
@@ -120,26 +120,26 @@ def accumulate_result(self, result, rerun=False):
         if not(result.state == State.CHILD_ERROR or result.state == State.INTERRUPTED) and not rerun:
             self.test_times.append((result.duration, test_name))

-        state = result.state
-        if state == State.PASSED:
-            self.good.append(test_name)
-        elif state == State.ENV_CHANGED:
-            self.environment_changed.append(test_name)
-        elif result.is_failed(self.ns.fail_env_changed):
-            if not rerun:
-                self.bad.append(test_name)
-                self.need_rerun.append(result)
-        elif state == State.SKIPPED:
-            self.skipped.append(test_name)
-        elif state == State.RESOURCE_DENIED:
-            self.skipped.append(test_name)
-            self.resource_denieds.append(test_name)
-        elif state == State.INTERRUPTED:
-            self.interrupted = True
-        elif state == State.DID_NOT_RUN:
-            self.run_no_tests.append(test_name)
-        else:
-            raise ValueError(f"invalid test state: {state!r}")
+        match result.state:
+            case State.PASSED:
+                self.good.append(test_name)
+            case State.ENV_CHANGED:
+                self.environment_changed.append(test_name)
+            case state if result.is_failed(self.ns.fail_env_changed):
+                if not rerun:
+                    self.bad.append(test_name)
+                    self.need_rerun.append(result)
+            case State.SKIPPED:
+                self.skipped.append(test_name)
+            case State.RESOURCE_DENIED:
+                self.skipped.append(test_name)
+                self.resource_denieds.append(test_name)
+            case State.INTERRUPTED:
+                self.interrupted = True
+            case State.DID_NOT_RUN:
+                self.run_no_tests.append(test_name)
+            case _:
+                raise ValueError(f"invalid test state: {state!r}")
```